core/test-extension/deployment/src/test/java/io/quarkus/naming/DisableJNDITestCase.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.naming;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.Hashtable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class DisableJNDITestCase {
+
+    @RegisterExtension
+    static QuarkusUnitTest unitTest = new QuarkusUnitTest();
+
+    @Test
+    public void testJNDIDisabled() throws Exception {
+        CompletableFuture<Boolean> cf = new CompletableFuture<>();
+        //we just want to test that there is a connection attempt
+        //we open a socket and will fail the test if anything actually manages
+        //to connect to it
+        try (ServerSocket s = new ServerSocket(1389)) {
+            Thread t = new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        s.accept().close();
+                        cf.complete(true);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            });
+            t.setDaemon(true);
+            t.start();
+            try {
+                new InitialContext().lookup("ldap://127.0.0.1:1389/a");
+            } catch (NamingException expected) {
+
+            }
+            try {
+                //now test explicitly setting the initial context
+                Hashtable<Object, Object> environment = new Hashtable<>();
+                environment.put(Context.INITIAL_CONTEXT_FACTORY, "javax.naming.ldap.InitialLdapContext");
+                new InitialContext(environment).lookup("ldap://127.0.0.1:1389/a");
+            } catch (NamingException expected) {
+
+            }
+            Assertions.assertThrows(TimeoutException.class, () -> {
+                cf.get(1, TimeUnit.SECONDS);
+            });
```

Comment:
Better use the AssertJ assertions here: 

```suggestion
            assertThatExceptionOfType(NamingException.class).isThrownBy(() -> new InitialContext().lookup("ldap://127.0.0.1:1389/a"));
            assertThatExceptionOfType(NamingException.class).isThrownBy(() -> {
                //now test explicitly setting the initial context
                Hashtable<Object, Object> environment = new Hashtable<>();
                environment.put(Context.INITIAL_CONTEXT_FACTORY, "javax.naming.ldap.InitialLdapContext");
                new InitialContext(environment).lookup("ldap://127.0.0.1:1389/a");
            });
            assertThatExceptionOfType(TimeoutException.class).isThrownBy(() -> cf.get(1, TimeUnit.SECONDS));
```